lib/ansible/modules/remote_mount_facts.py

```diff
@@ -0,0 +1,290 @@
+# Fedora, Debian, Ubuntu, Arch use /etc/mtab or /proc/mounts
+# Solaris uses /etc/mnttab
+
+
+TYPES = [
+    "cifs",
+    "ceph",
+    "fuse",
+    "glusterfs",
+    "gpfs",
+    "nfs",
+    "nfs4",
+    "vboxsf",
+    "zfs",
+]
+
+
+def get_mount_info(
+    module: AnsibleModule,
+    mount: str,
+    device: str,
+    uuids: dict[str, str],
+    udevadm_uuid: Callable[[str], str]
+) -> tuple[Optional[dict], Optional[str]]:
+    """
+    Attempts to get the mount size and UUID within the specified timeout.
+    """
+    seconds = module.params["timeout"]
+    on_error = module.params["timeout_handler"]
+
+    @timeout.timeout(seconds)
+    def _get_mount_info(mount, device, uuids, udevadm_uuid):
+        mount_size = get_mount_size(mount)
+        uuid = uuids.get(device, udevadm_uuid(device))
+        return mount_size, uuid
+
+    try:
+        return _get_mount_info(mount, device, uuids, udevadm_uuid)
+    except timeout.TimeoutError:
+        if on_error == "error":
+            module.fail_json(msg=f"{device} on {mount} timed out")
+        elif on_error == "warn":
+            module.warn(f"{device} on {mount} timed out")
+    return None, None
+
+
+def parse_mount(line: str) -> Optional[dict[str, str]]:
+    """
+    Parse a line of mount information and return a dictionary with mount details.
+    """
+    match = BSD_MOUNT_RE.match(line) or AIX_MOUNT_RE.match(line)
+    if match is None:
+        return None
+
+    mount_info = {"mount": match.group("mount")}
+    if len(match.groups()) == 3:
+        if "," in (fstype := match.group("fstype")):
+            fstype, options = fstype.split(", ", 1)
+        else:
+            options = ""
+        mount_info.update({
+            "fstype": fstype,
+            "options": options,
+            "device": match.group("device")
+        })
+    else:
+        device = match.group("node") + ":" + match.group("mounted_over")
+        mount_info.update({
+            "fstype": match.group("fstype"),
+            "options": match.group("options"),
+            "time": match.group("time"),
+            "device": device
+        })
+    return mount_info
+
+
+def parse_mount_output(module: AnsibleModule) -> list[dict[str, str]]:
+    """
+    Parse output from the mount command.
+    """
+    mount_path = module.get_bin_path("mount")
+    mount_output = ""
+    if mount_path:
+        rc, mount_out, err = module.run_command(mount_path)
+
+    mounts = []
+    for line in mount_out.splitlines():
+        if (mount_info := parse_mount(line)):
+            mounts.append(mount_info)
+    return mounts
+
+
+def list_current_mounts(module: AnsibleModule) -> list[dict[str, str]]:
+    """
+    Checks the following sources for current mounts:
+      * /etc/mtab
+      * /proc/mounts
+      * /etc/mnttab (Solaris)
+    If no information if found in the files, attempts to parse the output from mount.
+    """
+
+    mnttab = False
+    mtab_entries = ""
+    if os.path.exists("/etc/mtab"):
```

Comment:
no need for 'ifs', since get_file_content already checks:
```
    if os.path.exists(path) and os.access(path, os.R_OK):
```

Reply:
good spotting, refactored this function